metrics/prometheus.go

```diff
@@ -25,18 +25,36 @@ const (
 	promCustomSubsystem    = "custom"
 )
 
+const (
+	KiB = 1024
+	MiB = 1024 * KiB
+	GiB = 1024 * MiB
+)
+
+// headerSizeBuckets are chosen to cover typical max request header sizes:
+//   - 64 KiB for [AWS ELB](https://docs.aws.amazon.com/elasticloadbalancing/latest/userguide/how-elastic-load-balancing-works.html#http-header-limits)
+//   - 16 KiB for [NodeJS](https://nodejs.org/api/cli.html#cli_max_http_header_size_size)
+//   - 8 KiB for [Nginx](https://nginx.org/en/docs/http/ngx_http_core_module.html#large_client_header_buffers)
+//   - 8 KiB for [Spring Boot](https://docs.spring.io/spring-boot/appendix/application-properties/index.html#application-properties.server.server.max-http-request-header-size)
+var headerSizeBuckets = []float64{4 * KiB, 8 * KiB, 16 * KiB, 64 * KiB}
+
+// responseSizeBuckets are chosen to cover 2^(10*n) sizes up to 1 GiB and halves of those.
+var responseSizeBuckets = []float64{1, 512, 1 * KiB, 512 * KiB, 1 * MiB, 512 * MiB, 1 * GiB}
+
 // Prometheus implements the prometheus metrics backend.
 type Prometheus struct {
 	// Metrics.
 	routeLookupM               *prometheus.HistogramVec
 	routeErrorsM               *prometheus.CounterVec
 	responseM                  *prometheus.HistogramVec
+	responseSizeM              *prometheus.HistogramVec
 	filterCreateM              *prometheus.HistogramVec
 	filterRequestM             *prometheus.HistogramVec
 	filterAllRequestM          *prometheus.HistogramVec
 	filterAllCombinedRequestM  *prometheus.HistogramVec
-	proxyBackendM              *prometheus.HistogramVec
-	proxyBackendCombinedM      *prometheus.HistogramVec
```

Comment:
removed `proxy` prefix to make name consistent with the `Subsystem: promBackendSubsystem` option.